### PR TITLE
delete user by username

### LINKED ISSUE #19: Add Delete User By Name Endpoint
https://github.com/ariesprayoga/swagger-java-sample-app/blob/master/src/main/java/com/wordnik/swagger/sample/resource/UserResource.java

### PR DESCRIPTION
**Pull request by HubertTatar**
This closes #19. Current implementation deletes user from db.

### CODE REVIEW

File: functional_test/live_tests/users_test.py [hunk 1]
```diff
@@ -58,11 +58,16 @@ def test_update_user(pet_context, pet_store_client):
         "email": "wheresmycar@gmail.com",
         "password": "wickofyourwit",
         "phone": "215-789-0123",
-        "id": 1
+        "id": 2
```
Comment (pauljamescleary) [inline functional_test/live_tests/users_test.py]: Interesting, why did this have to be updated to 2?

**Comment by HubertTatar** [comment]
I was checking something and forgot to revert this to original state.
Reverted it back to one.

**Comment by pauljamescleary** [comment]
@HubertTatar thanks a lot for the contributions!  It is great to continue to make progress while work has taken up all my free time (and the Philadelphia Eagles).

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala [hunk 2]
```diff
@@ -65,11 +65,22 @@ class UserEndpoints[F[_]: Effect] extends Http4sDsl[F] {
         }
     }
 
+  private def deleteUserEndpoint(userService: UserService[F]): HttpService[F] =
+    HttpService[F] {
+      case DELETE -> Root / "users" / userByName =>
```
Comment (martin-g) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: s/userByName/userName/

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala [hunk 3]
```diff
@@ -65,11 +65,22 @@ class UserEndpoints[F[_]: Effect] extends Http4sDsl[F] {
...
+        } yield resp
+    }
+
+
   def endpoints(userService: UserService[F]): HttpService[F] =
     signupEndpoint(userService) <+>
     updateEndpoint(userService) <+>
     listEndpoint(userService)   <+>
-    searchByName(userService)
+    searchByName(userService)   <+>
```
Comment (martin-g) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala]: Not part of this PR but `searchByName` is not consistent with the others, it misses `Endpoint` suffix

File: src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/inmemory/UserRepositoryInMemoryInterpreter.scala [hunk 4]
```diff
@@ -31,6 +31,14 @@ class UserRepositoryInMemoryInterpreter[F[_]: Applicative] extends UserRepositor
 
   override def list(pageSize: Int, offset: Int): F[List[User]] =
     cache.values.toList.sortBy(_.lastName).slice(offset, offset + pageSize).pure[F]
+
+  override def deleteByUserName(userName: String): F[Option[User]] = {
+    val deleted = for {
+      user <- cache.values.find(u => u.firstName == userName)
```
Comment (martin-g) [inline src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/inmemory/UserRepositoryInMemoryInterpreter.scala]: s/firstName/userName/

File: src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpointsSpec.scala [hunk 5]
```diff
@@ -69,4 +69,63 @@ class UserEndpointsSpec
...
+
+  test("get user by userName") {
+    val userRepo = UserRepositoryInMemoryInterpreter[IO]()
+    val userValidation = UserValidationInterpreter[IO](userRepo)
+    val userService = UserService[IO](userRepo, userValidation)
+    val userHttpService: HttpService[IO] = UserEndpoints.endpoints[IO](userService)
+
+    implicit val userDecoder: EntityDecoder[IO, User] = jsonOf[IO, User]
+
+    val user = User("test", "test", "test", "test", "test", "test", None)
```
Comment (martin-g) [inline src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpointsSpec.scala]: The values are poor. If they were different it would catch the problem above with `firstName`

File: src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpointsSpec.scala [hunk 6]
```diff
@@ -69,4 +69,63 @@ class UserEndpointsSpec
...
+
+  test("delete user by userName") {
+    val userRepo = UserRepositoryInMemoryInterpreter[IO]()
+    val userValidation = UserValidationInterpreter[IO](userRepo)
+    val userService = UserService[IO](userRepo, userValidation)
+    val userHttpService: HttpService[IO] = UserEndpoints.endpoints[IO](userService)
+
+    implicit val userDecoder: EntityDecoder[IO, User] = jsonOf[IO, User]
+
+    val user = User("test", "test", "test", "test", "test", "test", None)
```
Comment (martin-g) [inline src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpointsSpec.scala]: see above

**Comment by HubertTatar** [comment]
@martin-g You are right. I will fix things you pointed out.
@pauljamescleary do I need to create new branch or can I restore this one and add another commit ?

**Comment by pauljamescleary** [comment]
@HubertTatar I would just submit a new PR, thanks!
